processor/src/decoder/tests.rs

```diff
@@ -1329,30 +1344,39 @@ fn dyn_block() {
     let join_addr = INIT_ADDR + EIGHT;
     check_op_decoding(&trace, 1, INIT_ADDR, Operation::Join, 0, 0, 0);
     // starting first span
-    let mul_basic_block_addr = join_addr + EIGHT;
+    let mstorew_basic_block_addr = join_addr + EIGHT;
     check_op_decoding(&trace, 2, join_addr, Operation::Span, 1, 0, 0);
-    check_op_decoding(&trace, 3, mul_basic_block_addr, Operation::Mul, 0, 0, 1);
-    check_op_decoding(&trace, 4, mul_basic_block_addr, Operation::End, 0, 0, 0);
+    check_op_decoding(&trace, 3, mstorew_basic_block_addr, Operation::MStoreW, 0, 0, 1);
+    check_op_decoding(&trace, 4, mstorew_basic_block_addr, Operation::End, 0, 0, 0);
     // starting second span
-    let save_basic_block_addr = mul_basic_block_addr + EIGHT;
-    check_op_decoding(&trace, 5, join_addr, Operation::Span, 1, 0, 0);
-    check_op_decoding(&trace, 6, save_basic_block_addr, Operation::MovDn4, 0, 0, 1);
-    check_op_decoding(&trace, 7, save_basic_block_addr, Operation::End, 0, 0, 0);
+    let push_basic_block_addr = mstorew_basic_block_addr + EIGHT;
+    check_op_decoding(&trace, 5, join_addr, Operation::Span, 2, 0, 0);
+    check_op_decoding(
+        &trace,
+        6,
+        push_basic_block_addr,
+        Operation::Push(Felt::from(FOO_ROOT_NODE_ADDR as u32)),
```

Comment:
nit: maybe it is worth defining something like a `push_42_op` variable so that we can use it here and also on line 1311 above.